### PR TITLE
Adding a package definition.

### PR DESCRIPTION
**Pull request by nikolay-mansas**
By github.com/nikolay-mansas

### CODE REVIEW

File: src/Packet.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    #         return self.unpacked_data['id']
+
+    def serialize(self) -> bytes:
+        return msgpack.packb(self.raw_data)
+
+    def to_model(self):
+        if self.raw_data.get('Request', None) is not None:
+            if self.q == "PingRequest":
+                return PingRequest(self)
+            elif self.q == "GetLastBlock":
```
Comment (DoctorEenot) [inline src/Packet.py:46]: GetLastBlockRequest

**Review by DoctorEenot — CHANGES_REQUESTED** [review]

File: src/Packet.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+                return GetLastBlock(self)
+            elif self.q == "NewBlockRequest":
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: NewBlock

File: src/Packet.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+    #         return self.unpacked_data['id']
+    #     else:
+    #         return self.unpacked_data['id']
+
+    def serialize(self) -> bytes:
+        return msgpack.packb(self.raw_data)
+
+    def to_model(self):
+        if self.raw_data.get('Request', None) is not None:
+            if self.q == "PingRequest":
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: Ping

File: src/Packet.py [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+                return PingRequest(self)
+            elif self.q == "GetLastBlock":
+                return GetLastBlock(self)
+            elif self.q == "NewBlockRequest":
+                return NewBlockRequest(self)
+            elif self.q == "GetNodesRequest":
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: GetNodes

File: src/Packet.py [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+                return GetNodesRequest(self)
+            elif self.q == "AnnounceRequest":
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: Announce

File: src/Packet.py [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+                return AnnounceRequest(self)
+            elif self.q == "GetAmountRequest":
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: GetAmount

File: src/Packet.py [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+                return GetAmountRequest(self)
+            elif self.q == "GetTransactionRequest":
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: GetTransaction

File: src/Packet.py [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+                return GetTransactionRequest(self)
+            elif self.q == "GetBlockByHashRequest":
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: GetBlockByHash

File: src/Packet.py [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+                return GetBlockByHashRequest(self)
+            elif self.q == "GetBlocksByHeightsRequest":
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: GetBlocksByHeights

File: src/Packet.py [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+                return GetBlocksByHeightsRequest(self)
+            elif self.q == "NewTransactionRequest":
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: NewTransaction

File: src/Packet.py [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+                return NewTransactionRequest(self)
+            elif self.q == "GetBlockByHeightRequest":
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: GetBlockByHeight

File: src/Packet.py [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+                return GetBlockByHeightRequest(self)
+            elif self.q == "GetBlocksByHeightsRequest":
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: Already exists at the line 60, remove this

File: src/Packet.py [hunk 13]
```diff
@@ -0,0 +1,214 @@
...
+                return GetBlocksByHeightsRequest(self)
+        elif self.raw_data.get('Response', None) is not None:
+            if self.q == "OkResponse":
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: Ok

File: src/Packet.py [hunk 14]
```diff
@@ -0,0 +1,214 @@
...
+                return OkResponse(self)
+            elif self.q == "PingResponse":
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: Ping

File: src/Packet.py [hunk 15]
```diff
@@ -0,0 +1,214 @@
...
+            if self.raw_data.get('Request', None) is None:
+                raise Exception("Not a request nor a response, bad packet")
+            self.is_request = True
+            self.unpacked_data = self.raw_data['Request']
+        else:
+            self.is_request = False
+            self.unpacked_data = self.raw_data['Response']
+
+        self.id = self.unpacked_data['id']
+        self.q = self.unpacked_data['q']
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: try:
    self.q = self.unpacked_data['q']
except KeyError:
    self.r = self.unpacked_data['r']

File: src/Packet.py (hunk 13 above)
Comment (YeahNotSewerSide) [inline src/Packet.py]: self.q -> self.r

File: src/Packet.py (hunk 14 above)
Comment (YeahNotSewerSide) [inline src/Packet.py]: self.q -> self.r

File: src/Packet.py [hunk 16]
```diff
@@ -0,0 +1,214 @@
...
+            elif self.q == "GetBlockByHeightRequest":
+                return GetBlockByHeightRequest(self)
+            elif self.q == "GetBlocksByHeightsRequest":
+                return GetBlocksByHeightsRequest(self)
+        elif self.raw_data.get('Response', None) is not None:
+            if self.q == "OkResponse":
+                return OkResponse(self)
+            elif self.q == "PingResponse":
+                return PingResponse(self)
+            elif self.q == "GetNodesResponse":
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: self.q -> self.r

Comment (YeahNotSewerSide) [inline src/Packet.py]: GetNodes

File: src/Packet.py [hunk 17]
```diff
@@ -0,0 +1,214 @@
...
+                return GetNodesResponse(self)
+            elif self.q == "GetBlockResponse":
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: self.q -> self.r

Comment (YeahNotSewerSide) [inline src/Packet.py]: GetBlock

File: src/Packet.py [hunk 18]
```diff
@@ -0,0 +1,214 @@
...
+                return GetBlockResponse(self)
+            elif self.q == "GetAmountResponse":
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: self.q -> self.r

Comment (YeahNotSewerSide) [inline src/Packet.py]: GetAmount

File: src/Packet.py [hunk 19]
```diff
@@ -0,0 +1,214 @@
...
+                return GetAmountResponse(self)
+            elif self.q == "SubmitPowResponse":
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: self.q -> self.r

Comment (YeahNotSewerSide) [inline src/Packet.py]: SubmitPow

File: src/Packet.py [hunk 20]
```diff
@@ -0,0 +1,214 @@
...
+                return SubmitPowResponse(self)
+            elif self.q == "GetBlocksResponse":
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: self.q -> self.r

Comment (YeahNotSewerSide) [inline src/Packet.py]: GetBlocks

File: src/Packet.py [hunk 21]
```diff
@@ -0,0 +1,214 @@
...
+                return GetBlocksResponse(self)
+            elif self.q == "GetTransactionResponse":
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: self.q -> self.r

Comment (YeahNotSewerSide) [inline src/Packet.py]: GetTransaction

File: src/Packet.py [hunk 22]
```diff
@@ -0,0 +1,214 @@
...
+        pass
+
+
+class PingRequest(Packet):
+    def __init__(self, pack: Packet):
+        super().__init__(pack.raw_data)
+        pass
+
+
+class GetLastBlock(Packet):
```
Comment (YeahNotSewerSide) [inline src/Packet.py]: GetLastBlockRequest

**Review by YeahNotSewerSide — CHANGES_REQUESTED** [review]

**Review by YeahNotSewerSide — APPROVED** [review]

**Pull request merged**